core/src/main/scala/org/apache/spark/executor/ExecutorSource.scala

```diff
@@ -73,6 +75,29 @@ class ExecutorSource(threadPool: ThreadPoolExecutor, executorId: String) extends
     registerFileSystemStat(scheme, "write_ops", _.getWriteOps(), 0)
   }
 
+  /** Dropwizard metrics gauge measuring the executor's process CPU time.
+   *  This code will try to get JVM Process CPU time or return -1 otherwise.
+   *  The CPU time value is returned in nanoseconds.
+   *  It will use proprietary extensions as com.sun.management.OperatingSystemMXBean or
+   *  com.ibm.lang.management.OperatingSystemMXBean if available
+   */
+  val mBean: MBeanServer = ManagementFactory.getPlatformMBeanServer
+  val name = new ObjectName("java.lang", "type", "OperatingSystem")
+  metricRegistry.register(MetricRegistry.name("executorCPUTime" ), new Gauge[Long] {
```

Comment:
a little confused with the exsiting `cpuTime`. How about `jvmCpuTime`?

Reply:
nit: `name("executorCPUTime" )` -> `name("executorCPUTime")`